infrastructure/templates/public-api/components/containerApp.bicep

```diff
@@ -54,87 +51,39 @@ param minReplica int = 1
+param appSettings {
+  name: string
+  @secure()
+  value: string
+}[]
 
 @description('A set of tags with which to tag the resource in Azure')
 param tagValues object
 
-@description('The Application Insights key that is associated with this resource')
-param applicationInsightsKey string
+@description('An existing Managed Identity\'s Resource Id with which to associate this Container App')
+param managedIdentityId string?
 
-var containerImageName = useDummyImage == true ? 'mcr.microsoft.com/azuredocs/aci-helloworld' : '${acrLoginServer}/${containerAppImageName}'
-var containerEnvName = '${resourcePrefix}-cae-${containerAppName}'
-var containerApplicationName = toLower('${resourcePrefix}-ca-${containerAppName}')
-var userIdentityName = '${resourcePrefix}-id-${containerAppName}'
-var containerLogName = '${resourcePrefix}-log-${containerAppName}'
-var acrPullRole = resourceId('Microsoft.Authorization/roleDefinitions', '7f951dda-4ed3-4680-a7ca-43fe172d538d')
+@description('Id of the owning Container App Environment')
+param managedEnvironmentId string
 
-resource logAnalytics 'Microsoft.OperationalInsights/workspaces@2021-06-01' = {
-  name: containerLogName
-  location: location
-  properties: {
-    sku: {
-      name: 'PerGB2018'
-    }
-  }
-  tags: tagValues
-}
-
-resource managedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' = {
-  name: userIdentityName
-  location: location
-}
-
-@description('This allows the managed identity of the container app to access the registry, note scope is applied to the wider ResourceGroup not the ACR')
-resource managedIdentityRBAC 'Microsoft.Authorization/roleAssignments@2022-04-01' = if (!useDummyImage) {
-  name: guid(resourceGroup().id, managedIdentity.id, acrPullRole)
-  properties: {
-    roleDefinitionId: acrPullRole
-    principalId: managedIdentity.properties.principalId
-    principalType: 'ServicePrincipal'
-  }
-}
+var containerImageName = '${acrLoginServer}/${containerAppImageName}'
+var containerApplicationName = toLower('${resourcePrefix}-ca-${containerAppName}')
 
-resource containerAppEnvironment 'Microsoft.App/managedEnvironments@2023-05-01' = {
-  name: containerEnvName
-  location: location
-  properties: {
-    vnetConfiguration: {
-      infrastructureSubnetId: subnetId
-    }
-    daprAIInstrumentationKey: applicationInsightsKey
-    appLogsConfiguration: {
-      destination: 'log-analytics'
-      logAnalyticsConfiguration: {
-        customerId: logAnalytics.properties.customerId
-        sharedKey: logAnalytics.listKeys().primarySharedKey
-      }
-    }
-    workloadProfiles: [
-      {
-        name: 'Consumption'
-        workloadProfileType: 'Consumption'
-      }
-    ]
+var containerAppIdentity = managedIdentityId != null ? {
+  type: 'UserAssigned'
+  userAssignedIdentities: {
+    '${managedIdentityId}': {}
   }
-  tags: tagValues
+} : {
+  type: 'SystemAssigned'
 }
 
 resource containerApp 'Microsoft.App/containerApps@2023-05-01' = {
```

Comment:
Was just curious - Have you folks considered putting a load balancer in front of the containers? Aware the usage that the public api is gonna get is a bit of an unknown but a load balancer would help with stampedes of traffic suddenly come thru to a given service. Also should provide lots of good things you can scale on out of the box i.e. if load balancer reqs > 10k then scale to 6 containers

Reply:
Well hello @luke-h1 :smile:  Hope you're doing well today!

I believe that Container Apps have built-in load balancing out of the box to a degree, but we'll need to see how that performs when all the code's good to go I guess.  The longer-term plans are still to have this sit behind Azure API Management for rate limiting, DDoS prevention etc, so this will likely add another dimension to this puzzle as well!

Reply:
Cool, yeah having a waf / the API management stuff will help against the nefarious traffic I imagine. Another advantage of the load balancer approach I forgot to mention is that your DNS is attached to the load balancer, rather than the container app/app service. Makes things like server patching/upgrades a bit easier as you don't have to go around and edit any DNS records 😛